File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples-js/Triggers_Create.js
const { DataFactoryManagementClient } = require("@azure/arm-datafactory");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Creates or updates a trigger.
 *
 * @summary Creates or updates a trigger.
 * x-ms-original-file: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/Triggers_Create.json
 */
async function triggersCreate() {
  const subscriptionId =
    process.env["DATAFACTORY_SUBSCRIPTION_ID"] || "12345678-1234-1234-1234-12345678abc";
  const resourceGroupName = process.env["DATAFACTORY_RESOURCE_GROUP"] || "exampleResourceGroup";
  const factoryName = "exampleFactoryName";
  const triggerName = "exampleTrigger";
  const trigger = {
    properties: {
      type: "ScheduleTrigger",
      pipelines: [
        {
          parameters: { outputBlobNameList: ["exampleoutput.csv"] },
          pipelineReference: {
            type: "PipelineReference",
            referenceName: "examplePipeline",
          },
        },
      ],
      recurrence: {
        endTime: new Date("2018-06-16T00:55:13.8441801Z"),
        frequency: "Minute",
        interval: 4,
        startTime: new Date("2018-06-16T00:39:13.8441801Z"),
        timeZone: "UTC",
      },
    },
  };
  const credential = new DefaultAzureCredential();
  const client = new DataFactoryManagementClient(credential, subscriptionId);
  const result = await client.triggers.createOrUpdate(
    resourceGroupName,
    factoryName,
    triggerName,
    trigger
  );
  console.log(result);
}

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples-python/Triggers_Create.py
from azure.identity import DefaultAzureCredential

from azure.mgmt.datafactory import DataFactoryManagementClient

"""
# PREREQUISITES
    pip install azure-identity
    pip install azure-mgmt-datafactory
# USAGE
    python triggers_create.py

    Before run the sample, please set the values of the client ID, tenant ID and client secret
    of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID,
    AZURE_CLIENT_SECRET. For more info about how to get the value, please see:
    https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal
"""


def main():
    client = DataFactoryManagementClient(
        credential=DefaultAzureCredential(),
        subscription_id="12345678-1234-1234-1234-12345678abc",
    )

    response = client.triggers.create_or_update(
        resource_group_name="exampleResourceGroup",
        factory_name="exampleFactoryName",
        trigger_name="exampleTrigger",
        trigger={
            "properties": {
                "pipelines": [
                    {
                        "parameters": {"OutputBlobNameList": ["exampleoutput.csv"]},
                        "pipelineReference": {"referenceName": "examplePipeline", "type": "PipelineReference"},
                    }
                ],
                "type": "ScheduleTrigger",
                "typeProperties": {
                    "recurrence": {
                        "endTime": "2018-06-16T00:55:13.8441801Z",
                        "frequency": "Minute",
                        "interval": 4,
                        "startTime": "2018-06-16T00:39:13.8441801Z",
                        "timeZone": "UTC",
                    }
                },
            }
        },
    )
    print(response)


# x-ms-original-file: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/Triggers_Create.json
if __name__ == "__main__":
    main()
